core/eolearn/core/eodata.py

```diff
@@ -44,6 +54,169 @@
+
+        if isinstance(value, FeatureIO) and load:
+            value = value.load()
+            self[feature_name] = value
+
+        return value
+
+    def __eq__(self, other: object) -> bool:
+        """Compares its content against a content of another feature type dictionary."""
+        return deep_eq(self, other)
+
+    def __ne__(self, other: object) -> bool:
+        """Compares its content against a content of another feature type dictionary."""
+        return not self.__eq__(other)
+
+    def get_dict(self) -> Dict[str, _T]:
+        """Returns a Python dictionary of features and value."""
+        return dict(self)
+
+    @abstractmethod
+    def _parse_feature_value(self, value: object, feature_name: str) -> _T:
+        """Checks if value fits the feature type. If not it tries to fix it or raise an error.
+
+        :raises: ValueError
+        """
+
+
+class _FeatureDictNumpy(_FeatureDict[np.ndarray]):
+    """_FeatureDict object specialized for Numpy arrays."""
+
+    def __init__(self, feature_dict: Dict[str, Union[np.ndarray, FeatureIO[np.ndarray]]], feature_type: FeatureType):
+        ndim = feature_type.ndim()
+        if ndim is None:
+            raise ValueError(f"Feature type {feature_type} does not represent a Numpy based feature.")
+        self.ndim = ndim
+        super().__init__(feature_dict, feature_type)
+
+    def _parse_feature_value(self, value: object, feature_name: str) -> np.ndarray:
+        if not isinstance(value, np.ndarray):
+            raise ValueError(f"{self.feature_type} feature has to be a numpy array.")
+        if not hasattr(self, "ndim"):  # Because of serialization/deserialization during multiprocessing
+            return value
+        if value.ndim != self.ndim:
+            raise ValueError(
+                f"Numpy array of {self.feature_type} feature has to have {self.ndim} "
+                f"dimension{'s' if self.ndim > 1 else ''} but feature {feature_name} has {value.ndim}."
+            )
+
+        if self.feature_type.is_discrete() and not is_discrete_type(value.dtype):
+            raise ValueError(
+                f"{self.feature_type} is a discrete feature type therefore dtype of data array "
+                f"has to be either integer or boolean type but feature {feature_name} has dtype {value.dtype.type}."
+            )
+
+        return value
+
+
+class _FeatureDictGeoDf(_FeatureDict[gpd.GeoDataFrame]):
+    """_FeatureDict object specialized for GeoDataFrames."""
+
+    def __init__(self, feature_dict: Dict[str, gpd.GeoDataFrame], feature_type: FeatureType):
+        if not feature_type.is_vector:
+            raise ValueError(f"Feature type {feature_type} does not represent a vector feature.")
+        super().__init__(feature_dict, feature_type)
+
+    def _parse_feature_value(self, value: object, feature_name: str) -> gpd.GeoDataFrame:
+        if isinstance(value, gpd.GeoSeries):
+            value = gpd.GeoDataFrame(dict(geometry=value), crs=value.crs)
```

Comment:
It turns out using `dict` here is unnecessary, we can simply do `gpd.GeoDataFrame(geometry=value, crs=value.crs)`. Sorry, this is my old mistake :see_no_evil:.